src/main/java/com/google/devtools/build/lib/rules/objc/ObjcCommon.java

```diff
@@ -661,6 +662,22 @@ static boolean shouldUseObjcModules(RuleContext ruleContext) {
     return false;
   }
 
+  static String getClangModuleName(RuleContext ruleContext) {
+    if (ruleContext.attributes().has("module_name", Type.STRING)) {
+      return ruleContext.attributes().get("module_name", Type.STRING);
```

Comment:
This user-supplied value should likely be validated against the restrictions that apply to module names. Normally we'd do this in the validate method(s) of the *Support class most related with the attribute but in this case the association isn't obvious (both bundling and compilation read it). I would say that this is more associated with compilation so validate the attribute there. In that case I'd also move this method there.